### ISSUE TITLE
Configurable roles for Page and PageHeader

### ISSUE THREAD

**Issue by seanforyou23** [patternfly/patternfly-react#2580]
Currently, these components default to `role="main"` and `role="banner"` respectively. These are sensible defaults, but shouldn't be hardcoded. We shouldn't assume that Patternfly layouts are always used at the root of an application. It's a feature that the page markup/styling is nestable, so we need to make sure users can customize these settings in these cases.

An immediate example is within our integration test application.

<img width="1110" alt="Screen Shot 2019-07-23 at 4 31 13 PM" src="https://user-images.githubusercontent.com/5942899/61745378-6f2d7e80-ad67-11e9-90d8-47abd4c0732d.png">

Generally, the banner role should only be used for the outermost header of an application. Similarly, there should only be a single main role and it should be the outermost container for the app's dynamic content.

**Comment by jenny-s51** [comment]
After corresponding with @seanforyou23 regarding this issue, he added some helpful feedback on what he meant by "configurable":

"We do want the `role` to be controllable by the user in case some advanced scenario comes up where people need more fine grain control over them"

I.e. the attributes can be removed, but add a "prop there as an option in case someone wants/needs to set it explicitly". This prop would set the `role` on either the `head` or the `main` element. 

I'd be happy to take on this issue once it's prioritized and I have some extra bandwidth 🙂
